### PR TITLE
replace timestampedCsv mode by csvOutput

### PR DESCRIPTION
**Pull request by mhdkassir**
## Summary
csvOutput mode saves the result in a csv file with the same name of the input file

**PR Type: Feature**

**Status: Ready To Review**

**Breaking change? No**

### How has this been tested?
- a local test for watch step with timestampedCsv option
- existing unit tests

### Other changes
- update an error message in DsvIngestionJob
- bugfix for pre and post encrypt generated yml

## Contributor checklist:
Go over all the following points, and put an `x` in all the boxes that apply.
If you're unsure about any of these, don't hesitate to ask. We're here to help!
- [x] My code follows the code style of this project.

### CODE REVIEW

File: src/main/scala/com/ebiznext/comet/job/ingest/IngestionJob.scala [hunk 1]
```diff
@@ -210,7 +214,7 @@ trait IngestionJob extends SparkJob {
         .head
       val finalCsvPath = new Path(
         acceptedPath,
-        s"${acceptedPath.getName()}-${now}.csv"
+        s"${acceptedPath.getName()}-${now}${path.headOption.map("-" + _.getName).getOrElse(".csv")}"
```
Comment (hayssams) [inline src/main/scala/com/ebiznext/comet/job/ingest/IngestionJob.scala:217]: Shouldn't we set the timestamp at the end of the filename ?

File: src/main/scala/com/ebiznext/comet/schema/generator/Xls2Yml.scala [hunk 2]
```diff
@@ -21,19 +21,23 @@ object Xls2Yml extends LazyLogging {
...
-            attr.copy(`type` = "string", required = false, privacy = None)
-        }
-      val newMetaData: Option[Metadata] = s.metadata.map { m => m.copy(partition = None) }
-      s.copy(attributes = newAtt)
+              attr.copy(`type` = "string", required = false)
+            else
+              attr.copy(`type` = "string", required = false, privacy = None)
+          }
+      val newMetaData: Option[Metadata] = s.metadata.map { m =>
+        m.copy(partition = None).copy(sink = None)
```
Comment (hayssams) [inline src/main/scala/com/ebiznext/comet/schema/generator/Xls2Yml.scala:38]: m.copy(partition = None, sink = None) would be simpler to read

File: src/main/scala/com/ebiznext/comet/schema/generator/Xls2Yml.scala [hunk 3]
```diff
@@ -21,19 +21,23 @@ object Xls2Yml extends LazyLogging {
...
+              )
             )
-          )
-            attr.copy(`type` = "string", required = false)
-          else
-            attr.copy(`type` = "string", required = false, privacy = None)
-        }
-      val newMetaData: Option[Metadata] = s.metadata.map { m => m.copy(partition = None) }
-      s.copy(attributes = newAtt)
+              attr.copy(`type` = "string", required = false)
```
Comment (hayssams) [inline src/main/scala/com/ebiznext/comet/schema/generator/Xls2Yml.scala:33]: I do not understand what is the purpose of this change

File: src/main/scala/com/ebiznext/comet/schema/generator/Xls2Yml.scala [hunk 4]
```diff
@@ -63,7 +67,8 @@ object Xls2Yml extends LazyLogging {
         metadata.copy(
           format = Some(Format.DSV),
           separator = delimiter.orElse(schema.metadata.flatMap(_.separator)).orElse(Some("µ")),
-          withHeader = schema.metadata.flatMap(_.withHeader)
+          withHeader = schema.metadata.flatMap(_.withHeader),
+          encoding = None
```
Comment (hayssams) [inline src/main/scala/com/ebiznext/comet/schema/generator/Xls2Yml.scala:71]: Is it required to set encoding to None

Comment (mhdkassir) [reply src/main/scala/com/ebiznext/comet/schema/generator/Xls2Yml.scala]: The encoding in the post-encrypt yml should be forced to UTF-8, in case we used a different encoding in preEncryption phase. I used None because UTF-8 is the default value. But you're right it is a bit confusing. fixed.

File: src/main/scala/com/ebiznext/comet/schema/generator/Xls2Yml.scala (hunk 2 above)
Comment (mhdkassir) [reply src/main/scala/com/ebiznext/comet/schema/generator/Xls2Yml.scala:38]: fixed

File: src/main/scala/com/ebiznext/comet/schema/generator/Xls2Yml.scala (hunk 3 above)
Comment (mhdkassir) [reply src/main/scala/com/ebiznext/comet/schema/generator/Xls2Yml.scala:33]: the main change is to ignore scripted attributes in the pre-encrypt yml (line 26), the rest is just spacing code format

File: src/main/scala/com/ebiznext/comet/job/ingest/IngestionJob.scala (hunk 1 above)
Comment (mhdkassir) [reply src/main/scala/com/ebiznext/comet/job/ingest/IngestionJob.scala]: No need for timestamp after all, this mode is renamed to csvOutput. The output file has the same name of the input file.

**Review by hayssams — APPROVED** [review]
